app/flux/API/ExerciseAPI.js

```diff
@@ -0,0 +1,77 @@
+var ExerciseStore = require('../stores/ExerciseStore');
+
+var exercise = {
+    get: {
+        exercise: (category, total, training, tree, type) => {
+            var options = { method: 'GET' };
+            var url = '/api/exercise';
+            
+            url = training ? url + '?trainingId=' + training : url;
+            url = category ? url + '&category=' + category : url;
+            url = tree ? url + '&tree=' + tree : url;
+            url = total ? url + '&total=' + total : url;
+            url = type ? url + '&type=' + type : url;
+
+            return fetch(url, options)
+                .then(response => {
+                    return response.json();
+                })
+                .then(response => {
+                    return Promise.resolve(response);
+                })
+                .catch(e => Promise.reject(e));            
+        }
+    },
+    put: {
+        exercise: () => {
+            var exerciseId = ExerciseStore.exercise.id;
+            var options = {
+                method: 'PUT',
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json; charset=utf-8'
+                }, 
+                body: JSON.stringify({
+                    id: exerciseId
+                })
+            };
+            
+            return fetch('/api/exercise', options)
+                .then(response => {
+                    return response.json();
+                })
+                .then(response => {
+                    return Promise.resolve(response);
+                })
+                .catch(e => Promise.reject(e));
+        },
+        question: (questionId, answer) => {
+            var exercise = ExerciseStore.exercise;
```

Comment:
Feels like an anti-pattern to be performing any logic here, seems that the exercise and question should just be passed in.